frontend/src/app/search-city.service.ts

```diff
@@ -13,19 +13,20 @@ export class SearchCityService {
 
   /**
    * Returns the first 10 cities matching a request
-   * @param term
-   * @returns {any}
    */
   search(term: string): Observable<Array<CityModel>> {
     if (term === '') {
       return Observable.of([]);
     }
 
-    const vicopoUrl = `${window.location.protocol}//vicopo.selfbuild.fr/cherche/${term}`;
-
-    return this.http
-      .get<any>(vicopoUrl)
-      .map(response => response.cities.slice(0, 10) as Array<CityModel>);
+    return this.http.get('/api/cities', { params : new HttpParams().set('query', term)});
```

Comment:
Indeed, but I made it generic enough: we don't send the token if the url starts with http. So it's valid for any external, present or future, external API. We'd better leave it there: it's just one line of code and it might still be helpgul later.